Removed deprecated `substr` string method with `substring` in history package

bilalk711 opened the pull request:
### Issue
In the `history` package, JS `substr` method was used that is deprecated now.

### Solution
I replaced that method with `substring` and the functionality is not affected. It is just a minor refactor.

bilalk711 commented:
@brophdawg11 can you please review and merge this PR?
It is just a minor refactor.

brophdawg11 commented:
@bilalk711 Could you clean up this PR so it's just your commits?  I'm seeing a bunch of unrelated commits in here.

Could you also add a changeset per https://github.com/remix-run/react-router/pull/12080#issuecomment-2395401315?

MichaelDeBoey commented on .changeset/silver-cats-shave.md:
```diff
@@ -0,0 +1,5 @@
+"react-router": patch
+---
+
+refactor(react-router): Replaced deprecated `substr` with `substring`in history package
```
```suggestion
Replaced `substr` with `substring`
```

bilalk711 commented:
> @bilalk711 Could you clean up this PR so it's just your commits? I'm seeing a bunch of unrelated commits in here.
> 
> Could you also add a changeset per [#12080 (comment)](https://github.com/remix-run/react-router/pull/12080#issuecomment-2395401315)?

I removed the irrelevant commits and added a changeset. 
I'm so sorry for that, I'll keep it in mind next time.

brophdawg11 reviewed (approved):
Thanks!

brophdawg11 commented:
Just FYI this is going into `dev` so it ill be included in the v7 release when that comes out
